src/data/index.ts

```diff
@@ -0,0 +1,16 @@
+export interface Course {
+	name: string;
+	crn?: number;
+	items: CourseItem[];
+}
+
+export interface CourseItem {
+	name: string;
+	weight: number;
+	grade?: number;
```

Comment:
I will add a due date, but should we possibly use a date type?[Date MDN]( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) I think the pros are that the month, day and year are part of separate properties and we can do things with formatting to display it how we want. If we use time as well then this is a pro, otherwise the time aspect is a con of complexity. String pros: relatively simple cons: we have to write our own parsing functions.

I don't have a preference, just wondering what you think or if you thought about this as well.